cs3api4lab/cs3_file_api.py

```diff
@@ -0,0 +1,237 @@
+"""
+cs3_file_api.py
+
+CS3 File API for the JupyterLab Extension
+
+Authors:
+"""
+
+import http
+import time
+
+import cs3.gateway.v1beta1.gateway_api_pb2 as cs3gw
+import cs3.gateway.v1beta1.gateway_api_pb2_grpc as cs3gw_grpc
+import cs3.rpc.code_pb2 as cs3code
+import cs3.storage.provider.v1beta1.provider_api_pb2 as cs3sp
+import cs3.storage.provider.v1beta1.resources_pb2 as cs3spr
+import grpc
+import requests
+
+
+class Cs3FileApi:
+    
+    tokens = {}  # map userid [string] to {authentication token, token expiration time}
+
+    log = None
+    chunksize = 4194304
+    authtokenvalidity = 3600
+    cs3_stub = None
+
+    client_id = None
+    client_secret = None
+
+    def __init__(self, config, log):
+
+        self.log = log
+        self.chunksize = int(config['chunksize'])
+        self.authtokenvalidity = int(config['authtokenvalidity'])
+
+        secure_channel = config['secure_channel']
+        reva_host = config['revahost']
+
+        self.client_id = config['client_id']
+        self.client_secret = config['client_secret']
+
+        # prepare the gRPC connection
+
+        if secure_channel:
+
+            try:
+
+                cert = open(config['client_cert'], 'rb').read()
+                key = open(config['client_key'], 'rb').read()
+
+                ca_cert = open(config['ca_cert'], 'rb').read()
+                credentials = grpc.ssl_channel_credentials(ca_cert, key, cert)
+
+                ch = grpc.secure_channel(reva_host, credentials)
+
+            except():
+                print("Error create secure channel")
+                raise
+
+        else:
+            ch = grpc.insecure_channel(reva_host)
+
+        self.cs3_stub = cs3gw_grpc.GatewayAPIStub(ch)
+        return
+
+    def _authenticate(self, userid):
```

Comment:
I will add the auth as a new issue. I would like to have a way to read the the oauth token from a file (because of the update process).